src/averageAges.js

```diff
@@ -37,9 +45,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
```

Comment:
The condition is correct, but consider adding a comment for better code readability, or maybe think about extracting it into a separate function